bin/src/ClassGenerator.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Graze\UnicontrollerClient\ClassGenerator;
+
+use Graze\UnicontrollerClient\ClassGenerator\Generator\GeneratorParser;
+use Graze\UnicontrollerClient\ClassGenerator\Generator\GeneratorEntity;
+use Graze\UnicontrollerClient\ClassGenerator\Generator\GeneratorSerialiser;
+
+class ClassGenerator
+{
+    /**
+     * @var GeneratorParser
+     */
+    private $generatorParser;
+
+    /**
+     * @var GeneratorEntity
+     */
+    private $generatorEntity;
+
+    /**
+     * @var GeneratorSerialiser
+     */
+    private $generatorSerialiser;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @param GeneratorParser $generatorParser
+     * @param GeneratorEntity $generatorEntity
+     * @param GeneratorSerialiser $generatorSerialiser
+     */
+    public function __construct(
+        GeneratorParser $generatorParser,
+        GeneratorEntity $generatorEntity,
+        GeneratorSerialiser $generatorSerialiser
+    ) {
+        $this->generatorParser = $generatorParser;
+        $this->generatorEntity = $generatorEntity;
+        $this->generatorSerialiser = $generatorSerialiser;
+    }
+
+    /**
+     * @param string $definition
+     * @param [] $arrayNameToItemName
+     */
+    public function generateClasses($definition, array $arrayNameToItemName)
+    {
+        // get the definition name
+        $pattern = '/^(?<name>[a-z]+)=/i';
+        $matches = [];
+        preg_match($pattern, $definition, $matches);
+
+        $this->name = $matches['name'];
+
+        // remove definition name
+        $definition = substr($definition, strlen($this->name)+1);
+
+        $properties = explode(',', $definition);
+        foreach ($properties as $property) {
+            // remove escape characters
+            $pattern = '/\[(STX|ETX)\]/';
+            $count = 0;
+            $property = preg_replace($pattern, '', $property, -1, $count);
+
+            $arrayItem = null;
+            $type = $count > 0 ? 'string' : 'int';
```

Comment:
This bit of the code takes command definitions copied and pasted from the API documentation. The definitions look like this - 
`[STX]filename[ETX],width,height,[STX]description[ETX]`
The 'STX' and 'ETX' are literal - there's no special meaning in the regex